app/uk/gov/hmrc/vatapi/orchestrators/VatReturnsOrchestrator.scala

```diff
@@ -88,11 +88,11 @@ trait VatReturnsOrchestrator extends ImplicitDateTimeFormatter {
 
   case class VatReturnOrchestratorResponse(nrs: NRSData, vatReturnResponse: VatReturnResponse)
 
-  private def buildNrsAudit(vrn: Vrn, nrsData: NRSData, request: Request[_]): AuditEvent[Map[String, String]] =
-    AuditEvents.nrsAudit(vrn, nrsData, request.headers.get("Authorization").getOrElse(""), request.headers.get("x-correlationid").getOrElse(""))
+  private def buildNrsAudit(vrn: Vrn, nrsData: NRSData, request: AuthRequest[_]): AuditEvent[Map[String, String]] =
+    AuditEvents.nrsAudit(vrn, nrsData, request.headers.get("Authorization").getOrElse(""))
 
-  private def buildSubmitVatReturnAudit(request: AuthRequest[_], nrSubmissionId: Option[String]): AuditEvent[Map[String, String]] =
-    AuditEvents.submitVatReturn(request.headers.get("x-correlationid").getOrElse(""), request.authContext.affinityGroup, nrSubmissionId)
+  private def buildSubmitVatReturnAudit(request: AuthRequest[_], response: VatReturnResponse, nrSubmissionId: Option[String]): AuditEvent[Map[String, String]] =
+    AuditEvents.submitVatReturn(response.underlying.header("CorrelationId").getOrElse(""), request.authContext.affinityGroup, nrSubmissionId)
```

Comment:
@MarkAKelly I didn't use `X-CorrelationId` because the header conversion that the response class does is only applied when the `filter` method is called on it - which is only done at the controller / resource level